### ISSUE TITLE
scrcpy: error while loading shared libraries: libavformat.so.58: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by bxt1nelox** [Genymobile/scrcpy#4538]
What am I supposed to do? Can you write the commands from start to finish, I am new to linux

**Comment by DerekWuYD** [comment]
This install command may help you.

```bash
sudo apt-get install libavformat58
```

**Comment by gharris999** [comment]
apt-get install libavformat58
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package libavformat58 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

**Comment by rom1v** [comment]
How did you install scrcpy?

**Comment by gharris999** [comment]
I don't use scrcpy. I ran into this problem trying to compile a different application (minidlnad) on a Ubuntu 23.10 box and found this thread while googling the issure.

**Comment by rom1v** [comment]
The reason (for both scrcpy or other software) is that you try to run a software built against libavformat58, but in your distrib, you now have `libavformat59` or `libavformat60`. You must re-build the software against the new libs.

**Comment by gharris999** [comment]
Yes, that's what I found in my case.  libavformat60 was installed and re-rebuilding minidlnad solved the problem.

**Comment by FibroJedi** [comment]
I know this thread is very old, but I am getting the libavformat.so.58 error.

I've updated via the command line (git pull, install_release etc).

Unfortunately, I don't know how to "rebuild" scrcpy. I'm not that kind of Linux user, ha. Can anyone direct me, please?
